applications/ChimeraApplication/custom_processes/apply_chimera_process.cpp

```diff
@@ -0,0 +1,627 @@
+        unsigned int start_constraint_id = i_bn * (TDim + 1) * (TDim + 1);
+        Element::Pointer r_host_element;
+        Vector weights;
+        bool is_found = SearchNode(rBinLocator, r_boundary_node, r_host_element, weights);
+        if (is_found) {
+            auto& ms_velocity_container =
+                rVelocityMasterSlaveContainerVector[omp_get_thread_num()];
+            auto& ms_pressure_container =
+                rPressureMasterSlaveContainerVector[omp_get_thread_num()];
+            removed_counter += RemoveExistingConstraintsForNode(r_boundary_node);
+            MakeConstraints(r_boundary_node, r_host_element, weights,
+                            ms_velocity_container, ms_pressure_container,
+                            constraints_id_vector, start_constraint_id);
+            found_counter += 1;
+        }
+    }
+
+    double loop_time = loop_over_b_nodes.ElapsedSeconds();
+    KRATOS_INFO_IF(
+        "ApplyChimera : Loop over boundary nodes took             : ", mEchoLevel > 0)
+        << loop_time << " seconds" << std::endl;
+    KRATOS_INFO_IF(
+        "ApplyChimera : Number of Boundary nodes                  : ", mEchoLevel > 1)
+        << n_boundary_nodes << std::endl;
+    KRATOS_INFO_IF(
+        "ApplyChimera : Number of Boundary nodes found            : ", mEchoLevel > 1)
+        << found_counter << std::endl;
+    KRATOS_INFO_IF(
+        "ApplyChimera : Number of Boundary nodes not found        : ", mEchoLevel > 1)
+        << n_boundary_nodes - found_counter << std::endl;
+    KRATOS_INFO_IF(
+        "ApplyChimera : Number of constraints made                : ", mEchoLevel > 1)
+        << found_counter * 9 << std::endl;
+    KRATOS_INFO_IF(
+        "ApplyChimera : Number of constraints removed             : ", mEchoLevel > 1)
+        << removed_counter << std::endl;
+}
+
+template <int TDim>
+ModelPart& ApplyChimera<TDim>::ExtractPatchBoundary(const Parameters PatchParameters,
+                                                    ModelPart& rBackgroundBoundaryModelpart,
+                                                    const ChimeraHoleCuttingUtility::Domain DomainType)
+{
+    Model& current_model = mrMainModelPart.GetModel();
+    const std::string patch_boundary_mp_name =
+        PatchParameters["boundary_model_part_name"].GetString();
+
+    if (!current_model.HasModelPart(patch_boundary_mp_name)) {
+        ModelPart& r_patch_model_part =
+            current_model.GetModelPart(PatchParameters["model_part_name"].GetString());
+        ModelPart& r_modified_patch_model_part =
+            r_patch_model_part.CreateSubModelPart(mModifiedName);
+        ModelPart& r_modified_patch_boundary_model_part =
+            r_modified_patch_model_part.CreateSubModelPart(
+                mBoundaryName + r_modified_patch_model_part.Name());
+        BuiltinTimer distance_calc_time_patch;
+        DistanceCalculationUtility<TDim>::CalculateDistance(
```

Comment:
I guess this is not the level set algorithm in the Kratos core? Am I wrong?

If so it is, why are you using an alternative one? Do you need any specific feature?